### ISSUE TITLE
Interface type option for routed networks

### ISSUE THREAD

**Issue by MetalJacx** [terraform-providers/terraform-provider-vcd#342]
Currently `vcd_network_routed` does not allow you to specify interface type and only deploys as an internal network. It would be nice to have a `interface_type =` option; that would default to internal.

![interface type](https://user-images.githubusercontent.com/13805973/65992817-da20c300-e444-11e9-87da-cafff48e6275.png)

```
resource "vcd_network_routed" "net" {
  org = "my-org" # Optional
  vdc = "my-vdc" # Optional

  name         = "my-net"
  edge_gateway = "Edge Gateway Name"
  gateway      = "10.10.0.1"
  interface_type = "distributed"    <------------ This Here.

  static_ip_pool {
    start_address = "10.10.0.152"
    end_address   = "10.10.0.254"
  }
}
```

**Comment by lvirbalas** [comment]
Joseph, thank you for a clean write-up. Appreciated. We're taking it in, unless you would be interested in contributing a PR? Our current release is fully booked, so there's time.

**Comment by MetalJacx** [comment]
@lvirbalas Thank You. I haven't done a PR, not sure the level I can contribute. Thank you for taking a look though.

**Comment by swerveshot** [comment]
This issue is partially a duplicate of #321 except this issue also describes the ability to use an internal interface type.

**Comment by dataclouder** [comment]
Implemented in #402
